spacy/util.py

```diff
@@ -949,130 +942,6 @@ def replace_model_node(model: Model, target: Model, replacement: Model) -> None:
-
-
-def run_command(
-    command: Union[str, List[str]],
-    *,
-    stdin: Optional[Any] = None,
-    capture: bool = False,
-) -> subprocess.CompletedProcess:
-    """Run a command on the command line as a subprocess. If the subprocess
-    returns a non-zero exit code, a system exit is performed.
-
-    command (str / List[str]): The command. If provided as a string, the
-        string will be split using shlex.split.
-    stdin (Optional[Any]): stdin to read from or None.
-    capture (bool): Whether to capture the output and errors. If False,
-        the stdout and stderr will not be redirected, and if there's an error,
-        sys.exit will be called with the return code. You should use capture=False
-        when you want to turn over execution to the command, and capture=True
-        when you want to run the command more like a function.
-    RETURNS (Optional[CompletedProcess]): The process object.
-    """
-    if isinstance(command, str):
-        cmd_list = split_command(command)
-        cmd_str = command
-    else:
-        cmd_list = command
-        cmd_str = " ".join(command)
-    try:
-        ret = subprocess.run(
-            cmd_list,
-            env=os.environ.copy(),
-            input=stdin,
-            encoding="utf8",
-            check=False,
-            stdout=subprocess.PIPE if capture else None,
-            stderr=subprocess.STDOUT if capture else None,
-        )
-    except FileNotFoundError:
-        # Indicates the *command* wasn't found, it's an error before the command
-        # is run.
-        raise FileNotFoundError(
-            Errors.E970.format(str_command=cmd_str, tool=cmd_list[0])
-        ) from None
-    if ret.returncode != 0 and capture:
-        message = f"Error running command:\n\n{cmd_str}\n\n"
-        message += f"Subprocess exited with status {ret.returncode}"
-        if ret.stdout is not None:
-            message += f"\n\nProcess log (stdout and stderr):\n\n"
-            message += ret.stdout
-        error = subprocess.SubprocessError(message)
-        error.ret = ret  # type: ignore[attr-defined]
-        error.command = cmd_str  # type: ignore[attr-defined]
-        raise error
-    elif ret.returncode != 0:
-        sys.exit(ret.returncode)
-    return ret
-
-
-@contextmanager
-def working_dir(path: Union[str, Path]) -> Iterator[Path]:
-    """Change current working directory and returns to previous on exit.
-
-    path (str / Path): The directory to navigate to.
-    YIELDS (Path): The absolute path to the current working directory. This
-        should be used if the block needs to perform actions within the working
-        directory, to prevent mismatches with relative paths.
-    """
-    prev_cwd = Path.cwd()
-    current = Path(path).resolve()
-    os.chdir(str(current))
-    try:
-        yield current
-    finally:
-        os.chdir(str(prev_cwd))
-
-
-@contextmanager
-def make_tempdir() -> Generator[Path, None, None]:
```

Comment:
I'm pretty sure we still need `make_tempdir` for the test suite. I guess you could move it into `spacy.tests.util`, but it seems fine to leave it here?

Reply:
There's a few more things that I need to fix, I'll take it out of review when I'm done!